pkgs/by-name/op/open-webui/package.nix

```diff
@@ -0,0 +1,116 @@
+{
+  lib,
+  buildNpmPackage,
+  fetchFromGitHub,
+  writeShellApplication,
+  python3,
+}:
+let
+  pname = "open-webui";
+  version = "0.1.124";
+
+  src = fetchFromGitHub {
+    owner = "open-webui";
+    repo = "open-webui";
+    rev = "v${version}";
+    hash = "sha256-r3oZiN2UIhPAG+ZcsZrXD1OemJrWXXlZdKVhK3+VhhU=";
+  };
+
+  frontend = buildNpmPackage {
+    inherit pname version src;
+
+    npmDepsHash = "sha256-uLp8QlPUR1dfchwu0IhJ8FFMMkm3V+FK2KBc41Un86g=";
+
+    env.CYPRESS_INSTALL_BINARY = "0"; # disallow cypress from downloading binaries in sandbox
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/share
+      cp -a build $out/share/open-webui
+
+      runHook postInstall
+    '';
+  };
+  pythonEnv = python3.withPackages (
+    p: with p; [
+      beautifulsoup4
+      gunicorn
+      fastapi
+      uvicorn
+      pydantic
+      python-multipart
+      flask
+      flask-cors
+      python-socketio
+      python-jose
+      passlib
+      requests
+      aiohttp
+      peewee
+      peewee-migrate
+      psycopg2
+      pymysql
+      bcrypt
+      litellm
+      boto3
+      argon2-cffi
+      apscheduler
+      google-generativeai
+      langchain
+      langchain-community
+      # langchain-chroma
+      fake-useragent
+      chromadb
+      sentence-transformers
+      pypdf
+      docx2txt
+      python-pptx
+      unstructured
+      markdown
+      pypandoc
+      pandas
+      openpyxl
+      pyxlsb
+      xlrd
+      validators
+      opencv4
+      # rapidocr-onnxruntime
+      fpdf2
+      rank-bm25
+      faster-whisper
+      pyjwt
+      black
+      # langfuse
+      youtube-transcript-api
+      pytube
+    ]
+  );
+in
+writeShellApplication {
```

Comment:
Instead of that, please use the PEP517 build process with `buildPythonApplication` and the `hatchling` build backend as seen here:

https://github.com/open-webui/open-webui/blob/72354e06a759075024d6be6bc6a8e717ec29d823/pyproject.toml#L85-L87

Reply:
Oh, thanks for pointing that out. I didn’t notice they had a `pyproject.toml`, I was just looking at the `backend` folder, which is why I didn’t go for `buildPythonApplication`.

Reply:
Done here: https://github.com/NixOS/nixpkgs/pull/316248/commits/76f247f67525dc076fe409b6d31364c640d5f127